CRM/MembershipExtras/Hook/BuildForm/UpdateSubscription.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/**
+ * Alters UpdateSubscription form.
+ */
+class CRM_MembershipExtras_Hook_BuildForm_UpdateSubscription {
+
+  private $form;
+  private $templatePath;
+  private $recurringContribution;
+
+  public function __construct(CRM_Contribute_Form_UpdateSubscription $form) {
+    $this->form = $form;
+    $this->templatePath = CRM_MembershipExtras_ExtensionUtil::path() . '/templates';
+
+    $recurringContributionID = CRM_Utils_Request::retrieve('crid', 'Integer', $this->form, FALSE);

```

Comment:
you can move the following two lines to a new method called : 

$this->setRecurringContribution()